UndertaleModLib/Models/UndertaleEmbeddedTexture.cs

```diff
@@ -311,62 +298,45 @@ public void Dispose()
     /// </summary>
     public class TexData : UndertaleObject, INotifyPropertyChanged, IDisposable
     {
-        private byte[] _textureBlob;
-        private static MemoryStream sharedStream;
+        private GMImage _image;
 
         /// <summary>
-        /// The image data of the texture.
+        /// The underlying image of the texture.
         /// </summary>
-        public byte[] TextureBlob
-        {
-            get => _textureBlob;
+        public GMImage Image 
+        { 
+            get => _image;
             set
             {
-                _textureBlob = value;
+                _image = value;
                 OnPropertyChanged();
             }
         }
 
         /// <summary>
         /// The width of the texture.
-        /// In case of an invalid texture data, this will be <c>-1</c>.
         /// </summary>
-        public int Width
-        {
-            get
-            {
-                if (_textureBlob is null || _textureBlob.Length < 24)
-                    return -1;
+        public int Width => _image.Width;
 
-                ReadOnlySpan<byte> span = _textureBlob.AsSpan();
-                return BinaryPrimitives.ReadInt32BigEndian(span[16..20]);
-            }
-        }
         /// <summary>
         /// The height of the texture.
-        /// In case of an invalid texture data, this will be <c>-1</c>.
         /// </summary>
-        public int Height
-        {
-            get
-            {
-                if (_textureBlob is null || _textureBlob.Length < 24)
-                    return -1;
-
-                ReadOnlySpan<byte> span = _textureBlob.AsSpan();
-                return BinaryPrimitives.ReadInt32BigEndian(span[20..24]);
-            }
-        }
+        public int Height => _image.Height;
 
         /// <summary>
         /// Whether this texture uses the QOI format.
         /// </summary>
-        public bool FormatQOI { get; set; } = false;
+        public bool FormatQOI => _image.Format is GMImage.ImageFormat.Qoi or GMImage.ImageFormat.Bz2Qoi;
 
         /// <summary>
         /// Whether this texture uses the BZ2 format. (Always used in combination with QOI.)
         /// </summary>
-        public bool FormatBZ2 { get; set; } = false;
+        public bool FormatBZ2 => _image.Format is GMImage.ImageFormat.Bz2Qoi;
+
+        /// <summary>
+        /// If located within a data file, this is the maximum end position of the image data (or start of the next texture blob).
```

Comment:
So the "maximum end position" in this context _is_ actually an upper bound on the end position of the image data. Inside a data file, it's either set to the start address of the next texture blob, or the end of the TXTR chunk if none exists. When loading textures, this is then narrowed down to the actual end address (but this field isn't updated).

I think I'm going to actually make this a private field, and just make a setter method for this, since this isn't meant to be relied upon for anything other than deserialization.

Reply:
I'm still confused by the name. If this is the end position of the image, why not just call it "endOfImagePosition"?

Reply:
I'm going to update this with a newer comment in just a moment, hopefully it'll be more clear - but this is the upper bound on the image end position, not the actual image end position.